nifi-nar-bundles/nifi-update-attribute-bundle/nifi-update-attribute-processor/src/main/java/org/apache/nifi/processors/attributes/UpdateAttribute.java

```diff
@@ -424,59 +419,104 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         // because is the original flowfile is used for all matching rules. in this
         // case the order of the matching rules is preserved in the list
         final Map<FlowFile, List<Rule>> matchedRules = new HashMap<>();
-        Map<String, String> statefulAttributes = null;
+
+        final Map<String, String> stateInitialAttributes;
+        final Map<String, String> stateWorkingAttributes;
+        StateMap stateMap = null;
 
         matchedRules.clear();
```

Comment:
not part of your changes but do we just want to be really sure the matched rules map instantiated above was empty?